web/src/components/SaveMessage.test.js

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+import { shallow } from "enzyme";
+import moment from "moment";
+import SaveMessage from "./SaveMessage";
+
+describe("<SaveMessage />", () => {
+  let lastSaved;
+  let subject;
+
+  describe('when saved less than 1 minute ago, it displays "Saved"', () => {
+    [
+      ["1 second ago", 1],
+      ["2 seconds ago", 2],
+      ["30 seconds ago", 30],
+      ["59 seconds", 59],
+    ].forEach(([testName, seconds]) => {
+      test(testName, () => {
+        lastSaved = moment().subtract(seconds, "seconds");
+        subject = shallow(
+          <SaveMessage lastSaved={lastSaved} />
+        );
+        expect(subject.text()).toEqual("Saved");
+      });
+    });
+  });
+
+  describe("when observed saved time changes to 1 minute ago", () => {
+    const now = new Date(2020, 0, 1, 12, 0);
+    const oneMinuteFromNow = new Date(2020, 0, 1, 12, 1);
+    let mockDateNow;
+
+    beforeEach(() => {
+      jest.useFakeTimers();
+      mockDateNow = jest
+        .spyOn(Date, "now")
+        .mockReturnValueOnce(now)
+        .mockReturnValue(oneMinuteFromNow);
+    });
+
+    afterEach(() => {
+      mockDateNow.mockRestore();
+      jest.clearAllTimers();
+    });
+
+    it('auto-updates', () => {
+      subject = shallow(
+        <SaveMessage isSaving={false} lastSaved={now} />
+      );
+      expect(subject.text()).toMatch("Saved");
+      jest.advanceTimersByTime(1000);
+      expect(subject.text()).toMatch(/\(1 minute ago\)$/);
+    });
+  });
+
+  describe("given current time is January 1, 2020 12:00 pm", () => {
+    const jan1AtNoon = new Date(2020, 0, 1, 12, 0);
+    let mockDateNow;
+
+    beforeEach(() => {
+      mockDateNow = jest
+        .spyOn(Date, "now")
+        .mockReturnValue(jan1AtNoon.getTime());
+    });
+
+    afterEach(() => {
+      mockDateNow.mockRestore();
+    });
+
+    describe("when saved 1 minute ago", () => {
+      it('displays "Last saved 11:59 am (1 minute ago)"', () => {
+        lastSaved = moment(jan1AtNoon).subtract(1, "minutes");
+        subject = shallow(
+          <SaveMessage lastSaved={lastSaved} />
+        );
+        expect(subject.text()).toMatch(/^Last saved/);
+        expect(subject.text()).toMatch(/11:59 am/);
+        expect(subject.text()).toMatch(/\(1 minute ago\)$/);
```

Comment:
I might want to condense these expectations to one line. I'm curious which you think is easier to read and easier to update.

```
expect(subject.text()).toMatch("Last saved 11:59 am (1 minute ago)");
```

Reply:
After that, it might be possible to format these remaining tests like on line 11

Reply:
Example: https://github.com/18F/cms-hitech-apd/compare/d91c94..ce556c

Reply:
I think I like them separate